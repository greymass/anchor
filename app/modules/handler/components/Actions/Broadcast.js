// @flow
import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { Button, Header, Icon, Label } from 'semantic-ui-react';

class PromptActionBroadcast extends Component<Props> {
  render() {
    const {
      disabled,
      loading,
      onClick,
      t,
      wallet,
    } = this.props;
    return (
      <Button
        animated="fade"
        as="div"
        disabled={disabled}
        floated="right"
        labelPosition="left"
        onClick={onClick}
      >
        <Label
          as="a"
          basic
          pointing="right"
        >
          <Header textAlign="left">
            <Header.Content>
              <Header.Subheader style={{ fontWeight: 'bold' }}>
                {t('handler_action_broadcast_header')}
              </Header.Subheader>
              {wallet.account}@{wallet.authorization}
            </Header.Content>
          </Header>
        </Label>
        <Button
          color="purple"
          icon
          loading={loading}
          style={{ position: 'relative' }}
        >
          <Button.Content hidden style={{ marginTop: '-0.75em' }}>
            <Icon
              name="cloud upload"
              size="large"
              style={{ marginRight: 0 }}
            />
          </Button.Content>
          <Button.Content visible style={{ margin: '0 1em' }}>
            <Icon
              name="cloud"
              size="large"
              style={{ marginRight: 0 }}
            />
          </Button.Content>
        </Button>
      </Button>
    );
  }
}

export default withTranslation('handler')(PromptActionBroadcast);
