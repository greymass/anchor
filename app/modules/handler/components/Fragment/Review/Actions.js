// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Trans, withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import { Grid, Header, Segment, Tab } from 'semantic-ui-react';

import PromptReviewControls from '../../../components/Review/Controls';
import ErrorMessage from '../../../components/error';
import PromptFragmentPlaceholderTransactionAction from '../../../components/Fragment/Placeholder/Transaction/Action';
import PromptFragmentTransactionAction from '../../../components/Fragment/Transaction/Action';
import PromptFragmentTransactionActionResourceProvider from '../../../components/Fragment/Transaction/Action/ResourceProvider';

class PromptFragmentReviewActions extends Component<Props> {
  render() {
    const {
      enableWhitelist,
      error,
      hasResourceProviderFee,
      loading,
      modifyWhitelist,
      t,
      transaction,
      usingResourceProvider,
      whitelist,
    } = this.props;
    return (
      <React.Fragment>
        {(loading)
          ? <PromptFragmentPlaceholderTransactionAction />
          : false
        }
        {(transaction)
          ? transaction.actions.map((action, index) => (
            <PromptFragmentTransactionAction
              action={action}
              enableWhitelist={enableWhitelist}
              modifyWhitelist={modifyWhitelist}
              index={index}
              showAll
              total={transaction.actions.length}
              whitelist={whitelist}
            />
          ))
          : false
        }
        {(error)
          ? (
            <Segment attached>
              <ErrorMessage
                error={error}
              />
            </Segment>
          )
          : false
        }
      </React.Fragment>
    );
  }
}

export default withTranslation('handler')(PromptFragmentReviewActions);
