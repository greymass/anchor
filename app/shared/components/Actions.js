// @flow
import React, { Component } from 'react';
import { Header, Grid, Loader, Segment, Visibility } from 'semantic-ui-react';
import { translate } from 'react-i18next';

import SidebarConnection from '../containers/Sidebar/Connection';
import ActionsTable from './Actions/Table';
import WalletPanel from './Wallet/Panel';

type Props = {
  actions: {
    clearSystemState: () => void,
    getAccount: () => void,
    getAccountActions: () => void
  },
  actionObjects: {},
  accounts: {},
  balances: {},
  history: {},
  keys: {},
  settings: {},
  system: {},
  t: () => void,
  actionObjects: {},
  transaction: {},
  validate: {},
  wallet: {}
};

class Actions extends Component<Props> {
  props: Props;

  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.tick();
    this.interval = setInterval(this.tick.bind(this), 15000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  tick() {
    const {
      actions,
      validate
    } = this.props;
    const {
      getAccountActions
    } = actions;
    if (validate.NODE) {
      getAccountActions();
    }
  }

  render() {
    const {
      actions,
      actionObjects,
      accounts,
      balances,
      history,
      keys,
      settings,
      system,
      t,
      transaction,
      validate,
      wallet
    } = this.props;

    const amount = 20;

    let sidebar = [(
      <WalletPanel
        actions={actions}
        accounts={accounts}
        balances={balances}
        key="WalletPanel"
        keys={keys}
        settings={settings}
        system={system}
        transaction={transaction}
        validate={validate}
        wallet={wallet}
      />
    )];

    const validUser = (keys && keys.key);

    if (validUser) {
      sidebar = (
        <React.Fragment />
      );
    }

    return (
      <div ref={this.handleContextRef}>
        <Grid divided>
          <Grid.Row>
            <Grid.Column width={6}>
              <SidebarConnection
                actions={actions}
                history={history}
              />
              {sidebar}
            </Grid.Column>
            <Grid.Column width={10}>
              {(keys.key && actionObjects.list.length > 0)
               ? [(
                 <Visibility
                   continuous
                   key="ActionsTable"
                   fireOnMount
                   onBottomVisible={this.loadMore}
                   once={false}
                 >
                   <ActionsTable
                     amount={amount}
                     attached="top"
                     actionObjects={actionObjects}
                   />
                 </Visibility>
               ), (
                 (amount < actionObjects.list.length)
                 ? (
                   <Segment key="ActionsTableLoading" clearing padded vertical>
                     <Loader active />
                   </Segment>
                 ) : false
               )]
               : (
                 <Segment attached="bottom" stacked>
                   <Header textAlign="center">
                     {t('actions_need_to_enter_password')}
                   </Header>
                 </Segment>
               )
              }
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}

export default translate('actions')(Actions);
