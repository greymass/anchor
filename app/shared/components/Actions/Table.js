// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Segment, Transition, Table } from 'semantic-ui-react';

import ActionsTableRow from './Table/Row';

class ActionsTable extends Component<Props> {
  render() {
    const {
      amount,
      actions,
      t
    } = this.props;

    const loading = (actions.list.length < 1);
    let baseTable = <Table.Body />;
    if (!loading) {
      const fullResults = actions.list.slice(0, amount);
      baseTable = (
        <Table.Body key="FullResults">
          {fullResults.map((action) => (
            <ActionsTableRow
              key={action.key}
              action={action}
            />
          ))}
        </Table.Body>
      );
    }
    return (
      <Segment basic loading={loading} vertical>
        <Table
          color="violet"
          size="small"
          striped
          style={{ borderRadius: 0 }}
          unstackable
        >
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell collapsing />
              <Table.HeaderCell collapsing />
              <Table.HeaderCell>
                {t('action_type')}
              </Table.HeaderCell>
              <Table.HeaderCell width={5}>
                {t('action_time')}
              </Table.HeaderCell>
              <Table.HeaderCell collapsing />
            </Table.Row>
          </Table.Header>
          <Transition animation="slide down" duration={200}>
            {baseTable}
          </Transition>
        </Table>
      </Segment>
    );
  }
}

export default translate('actions')(ActionsTable);
