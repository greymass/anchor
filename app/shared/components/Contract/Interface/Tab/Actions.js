// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Segment } from 'semantic-ui-react';

import ContractInterfaceSelectorAction from '../Selector/Action';
import ContractInterfaceFormAction from '../Form/Action';

class ContractInterfaceTabActions extends Component<Props> {
  render() {
    const {
      actions,
      blockExplorers,
      contract,
      contractAction,
      onChange,
      onSubmit,
      settings,
      system,
      transaction
    } = this.props;
    return (
      <React.Fragment>
        <ContractInterfaceSelectorAction
          contract={contract}
          contractAction={contractAction}
          onChange={onChange}
          onSubmit={onSubmit}
        />
        {(contractAction)
          ? (
            <Segment secondary>
              <ContractInterfaceFormAction
                actions={actions}
                blockExplorers={blockExplorers}
                contract={contract}
                contractAction={contractAction}
                settings={settings}
                system={system}
                transaction={transaction}
              />
            </Segment>
          ) : false
        }
      </React.Fragment>
    );
  }
}

export default translate('contract')(ContractInterfaceTabActions);
