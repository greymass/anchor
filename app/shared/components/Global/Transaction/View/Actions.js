// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Header } from 'semantic-ui-react';

import GlobalTransactionViewAction from './Action';

type Props = {
  actions: [],
  t: {}
};

class GlobalTransactionViewActions extends Component<Props> {
  props: Props;
  render() {
    const {
      actions,
      t
    } = this.props;
    return (
      <React.Fragment>
        <Header>
          {t('global_transaction_actions_details_title')}
          <Header.Subheader>
            {t('global_transaction_actions_details_content')}
          </Header.Subheader>
        </Header>
        {actions.map((action) => (
          <GlobalTransactionViewAction
            action={action}
          />
        ))}
      </React.Fragment>
    );
  }
}

export default translate('global')(GlobalTransactionViewActions);
