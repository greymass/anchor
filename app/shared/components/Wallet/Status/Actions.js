// @flow
import React, { Component } from 'react';
import { Header, Grid, Loader, Segment, Visibility } from 'semantic-ui-react';
import { translate } from 'react-i18next';

import ActionsTable from './Actions/Table';

type Props = {
  actions: {
    getAccountActions: () => void
  },
  actionObjects: {},
  t: () => void,
  validate: {}
};

class Actions extends Component<Props> {
  props: Props;

  constructor(props) {
    super(props);
    this.state = {
      amount: 20
    };
  }

  componentDidMount() {
    this.tick();
    this.interval = setInterval(this.tick.bind(this), 15000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  loadMore = () => this.setState({ amount: this.state.amount + 20 }, () => {
    this.tick();
  });

  tick() {
    const {
      actions,
      validate
    } = this.props;
    const {
      getAccountActions
    } = actions;

    if (validate.NODE) {
      const {
        amount
      } = this.state;

      getAccountActions(amount);
    }
  }

  render() {
    const {
      actionObjects,
      t,
    } = this.props;

    const {
      amount
    } = this.state;

    return (
      <div ref={this.handleContextRef}>
        <Grid.Column width={10}>
          {(actionObjects.list.length > 0)
           ? [(
             <Visibility
               continuous
               key="ActionsTable"
               fireOnMount
               onBottomVisible={this.loadMore}
               once={false}
             >
               <ActionsTable
                 amount={amount}
                 attached="top"
                 actionObjects={actionObjects}
               />
             </Visibility>
           ), (
             (amount > actionObjects.list.length)
             ? (
               <Segment key="ActionsTableLoading" clearing padded vertical>
                 <Loader active />
               </Segment>
             ) : false
           )]
           : (
             <Segment attached="bottom" stacked>
               <Header textAlign="center">
                 {t('actions_none')}
               </Header>
             </Segment>
           )
          }
        </Grid.Column>
      </div>
    );
  }
}

export default translate('actions')(Actions);
