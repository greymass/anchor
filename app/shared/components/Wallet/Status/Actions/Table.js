// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Segment, Transition, Table } from 'semantic-ui-react';

import ActionsTableRow from './Table/Row';

class WalletStatusActionsTable extends Component<Props> {
  render() {
    const {
      amount,
      actionObjects,
      settings,
      t
    } = this.props;

    const loading = (actionObjects.list.length < 1);
    let baseTable = <Table.Body />;
    if (!loading) {
      const fullResults = actionObjects.list.slice(0, amount);
      baseTable = (
        <Table.Body key="FullResults">
          {fullResults.map((actionObject) => (
            <ActionsTableRow
              actionObject={actionObject}
              key={actionObject.account_action_seq}
              settings={settings}
            />
          ))}
        </Table.Body>
      );
    }
    return (
      <Segment basic loading={loading} vertical>
        <Table
          color="violet"
          size="small"
          striped
          style={{ borderRadius: 0 }}
          unstackable
        >
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell width={6}>
                {t('actions_name')}
              </Table.HeaderCell>
              <Table.HeaderCell width={2}>
                {t('actions_time')}
              </Table.HeaderCell>
              <Table.HeaderCell width={2} />
            </Table.Row>
          </Table.Header>
          <Transition animation="slide down" duration={200}>
            {baseTable}
          </Transition>
        </Table>
      </Segment>
    );
  }
}

export default translate('actions')(WalletStatusActionsTable);
