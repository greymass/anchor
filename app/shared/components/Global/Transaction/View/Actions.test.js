import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionViewActions } from './Actions';

const mockProps = {
  actions: [],
  t: text => text,
};

describe('GlobalTransactionViewActions', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionViewActions {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

