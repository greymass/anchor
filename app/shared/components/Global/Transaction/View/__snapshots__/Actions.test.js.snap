// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionViewActions renders correctly 1`] = `
<Fragment>
  <Header>
    global_transaction_actions_details_title
    <HeaderSubheader>
      global_transaction_actions_details_content
    </HeaderSubheader>
  </Header>
</Fragment>
`;
